src/ripple/basics/SlabAllocator.h

```diff
@@ -57,40 +57,66 @@ class SlabAllocator
         // A linked list of appropriately sized free buffers:
         std::uint8_t* l_ = nullptr;
 
-        // The next memory block
+        // The next block:
         SlabBlock* next_;
 
         // The underlying memory block:
-        std::uint8_t const* const p_ = nullptr;
+        std::atomic<std::uint8_t*> data_ = nullptr;
 
-        // The extent of the underlying memory block:
-        std::size_t const size_;
+        // The number of items allocated from this block
+        std::atomic<std::size_t> avail_{0};
```

Comment:
`avail_` is always accessed within a lock so I don't think this needs to be `atomic`